### ISSUE TITLE
Mass Move -- Setting Change

### ISSUE THREAD

**Issue by jun6lee** [mylar3/mylar3#479]
Firstly I've only recently installed this and am kicking myself for not doing so sooner. Kudos for the efforts.

So I've been manually adding each series that I have completed to build my library, but a 100 or so later am coming to realise that the comics folder I've assigned is going to be humungous with all the one-shots taking up a directory each.

Not an issue, I've now gone the $Publisher/ way to sort that out going forwards.

**The Enhancement:**
Can I now correct the 100+ that I've added and have them moved to the correct folders?
Eg: "/comics/Ultimate Adventures (2002)" to "/comics/Marvel/Ultimate Adventures (2002)"

Just with the Ultimate Universe, this is seeming like a dawning task, never mind the rest of them. ;)

Sure, I can just move the files, but that way they'll all appear "archive" state.
What I want is them to stay in the "Downloaded" state so that I can open them directly through Mylar.

**Comment by evilhero** [comment]
There isn't really a way to mass update the directory locations within Mylar and at the same time rename the existing directory structures. You can mass update the directory locations, but even if the directories are created after you would still have to move the files into them from the old locations which can get confusing if you have a large number of files.

But what you can do, and I think will work:
- create the publishers folder in the /comics/ directory so that you can physically move the files later.
- use the locmove aspect (I'll explain in a bit) to change the series locations in the DB to the new locations. Once this is done, shut down Mylar.
- move your files based on publisher into the new location publisher folder (move the entire series folders intact).
- start up Mylar and the files should all be in a downloaded state still and I think can be properly referenced again.
- if they can't be located - a simple Recheck Files (or even a Refresh Series) will line everything back up again. And yes you can do this in a batch mode by going into the Manage tab / Manage Comics option.

The locmove option is an option whereby enabling it in the config.ini and starting up Mylar (since you can't edit the ini with Mylar running), will change the locations of all the series on your watchlist to a designated new location while keeping the folder formatting previously. Here's the breakdown:

- backup your mylar.db (just in case).

- edit your config.ini. These are the settings that will be used/changed:
```
[General]
destination_dir = /some/directory/structure

[Update]
newcom_dir = ""
fftonewcom_dir = False
locmove = False
```

**For this to work you need to keep the destination_dir (Comic Location in the GUI) pointed to the original location**. 
Once the changes have been made and the database changes are completed, and verified by yourself - then in the GUI change the Comic Location to the new path, save and then restart Mylar.

Setting ```locmove = True``` will enable the one time mass update of each series' Comic Location. 

If you leave ```newcom_dir``` blank (or as is "") this entire process will not do anything. 

Setting a path (with no trailing slash!) will update every and all series to the ```newcom_dir``` path but keep the existing folder name for the given series that is already present. As an example, if :
```
destination_dir = /mnt/evilhero/Comics
newcom_dir = /mnt/Comics

Upon the next startup/restart it will update the series location for every series in your watchlist in Mylar from /mnt/evilhero/Comics</series> to /mnt/Comics/<series>.
Once finished, Mylar will then reset the locmove flag to Fasle so it doesn't try to run on every startup thereafter.
```
fftonewcom_dir (True/False). Enabling this option will apply your current Folder Format string to every directory update in the db that's being done at the same time as it's converting directories. This also includes subdirectories within the Folder format (ie. ```$Publisher/$Series/$Volume ($Year)```) - so if you're wanting to move from a general format (ie. /mnt/Comics/Series) to something a bit more organized (ie. /mnt/Comics/Publisher/Series), you would enable this while making sure your Folder Format is set to the appropriate format. 

Again this doesn't physically touch your files on the system, it only tells Mylar to monitor the given series at the new specified location (and if it doesn't exist when it goes to check, it will create it).

And once this is all done and you've verified that everything is pointing to the new locations, make sure to update your Comic Location path in the GUI (or destination_dir in the ini) to the new location root. Otherwise, any new additions or even when a series refreshes will change the structure to the old location.

That's a blurb and half - any Q's just throw them this way.

**Comment by lamahmud** [comment]
I get an error when I attempt to do this.
12-Oct-2020 13:50:49 - DEBUG :: mylar.initialize.247 : MainThread : SEARCH_TIER_DATE set to : 2020-09-28
12-Oct-2020 13:50:49 - INFO :: mylar.updateComicLocation.812 : MainThread : Performing a one-time mass update to Comic Location
12-Oct-2020 13:50:49 - INFO :: mylar.validateAndCreateDirectory.1723 : MainThread : [DIRECTORY-CHECK] Found comic directory: /Volumes/Plex8/OneDrive/NewComics
12-Oct-2020 13:50:49 - ERROR :: mylar.excepthook.315 : MainThread : Uncaught exception: Traceback (most recent call last):
  File "Mylar.py", line 386, in <module>
    main()
  File "Mylar.py", line 203, in main
    mylar.initialize(mylar.CONFIG_FILE)
  File "/Volumes/MacHD/Users/lam/Applications/mylar3/mylar/__init__.py", line 260, in initialize
    helpers.updateComicLocation()
  File "/Volumes/MacHD/Users/lam/Applications/mylar3/mylar/helpers.py", line 889, in updateComicLocation
    com_done = re.sub('%&', os.sep.encode('unicode-escape'), comlocation).strip()
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: sequence item 0: expected str instance, bytes found

Traceback (most recent call last):
  File "Mylar.py", line 386, in <module>
    main()
  File "Mylar.py", line 203, in main
    mylar.initialize(mylar.CONFIG_FILE)
  File "/Volumes/MacHD/Users/lam/Applications/mylar3/mylar/__init__.py", line 260, in initialize
    helpers.updateComicLocation()
  File "/Volumes/MacHD/Users/lam/Applications/mylar3/mylar/helpers.py", line 889, in updateComicLocation
    com_done = re.sub('%&', os.sep.encode('unicode-escape'), comlocation).strip()
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: sequence item 0: expected str instance, bytes found

**Comment by lamahmud** [comment]
I'm attempting to move my file locations within the DB from /Volumes/Plex4/OneDrive/NewComics to a new hard drive which is /Volumes/Plex8/OneDrive/NewComics.

My config.ini lines are:
destination_dir = /Volumes/Plex8/OneDrive/NewComics
locmove = True
newcom_dir = /Volumes/Plex8/OneDrive/NewComics
fftonewcom_dir = False

**Comment by evilhero** [comment]
@lamahmud , we would prefer if this convo goes longer than my response that you open a new issue with the relevant details as this is getting away from the OG poster's intent.

However, as per the above instructions: 
```For this to work you need to keep the destination_dir (Comic Location in the GUI) pointed to the original location.```, which it is not in your blurb above.
